tests/run/analyse_expressions.srctree

```diff
@@ -0,0 +1,44 @@
+# mode: run
+# tag: coverage
+
+PYTHON setup.py build_ext --inplace
+
+######## moda.pyx ########
+
+cpdef mod_name(self):
+    return "mod A"
+
+cdef class ClassA:
+    cpdef class_name(self):
+        return "Class A"
+
+cdef class ClassB(ClassA):
+    cpdef class_name(self):
+        return "Class B"
+
+######## modb.pyx ########
+
+import moda
+
+cpdef mod_name_override(self):
+    return "I was overridden"
+
+moda.mod_name = mod_name_override
+
+######## setup.py ########
+
+from setuptools import setup
+from Cython.Build import cythonize
+import Cython.Compiler.Options
+
+Cython.Compiler.Options.lookup_module_cpdef = True
+
+setup(
+        ext_modules = cythonize(["moda.pyx", "modb.pyx"],
+            compiler_directives={'language_level': 3})
+)
+
+import moda, modb
+
+assert moda.mod_name == modb.mod_name_override
+assert moda.ClassA.class_name != moda.ClassB.class_name
```

Comment:
```suggestion
######## main.py ########

import moda, modb

assert moda.mod_name == modb.mod_name_override
assert moda.ClassA.class_name != moda.ClassB.class_name
```

1. It's a little odd running the test code in `setup.py` so I've suggested making a new test file to do it.
2. Can we call and run these methods too?

    ```
    assert moda._mod_name() == "I was overridden"
    ```